errors/errors.go

```diff
@@ -24,16 +28,28 @@ const (
 
 // BoulderError represents internal Boulder errors
 type BoulderError struct {
-	Type   ErrorType
-	Detail string
+	Type      ErrorType
+	Detail    string
+	SubErrors []SubBoulderError
+}
+
+// TODO(@cpu): SubBoulderError type docstring
+type SubBoulderError struct {
+	BoulderError
+	Identifier identifier.ACMEIdentifier
 }
 
-func (be *BoulderError) Error() string {
+func (be BoulderError) Error() string {
 	return be.Detail
 }
 
+func (be *BoulderError) WithSubErrors(subErrs []SubBoulderError) *BoulderError {
```

Comment:
Initially I was opposed to using this "builder" syntax as opposed to updating `New` and each of the convenience functions (`InternalServerError`, `MalformedError`, etc) to accept a `subErrs []SubBoulderError` argument. I ended up back on this approach because the variadic `... interface{}` syntax used by the convenience functions means the `subErrs` arg has to be somewhere earlier in the function signature and it felt weird to use `(type, subErrs, msg, args)`. Extending the function arguments also meant updating **all** of the convenience functions for consistency when in reality its unlikely that a `OrderNotReadyError` would have suberrors or benefit from the new arg.